tests/integration/test_charm_metrics_failure.py

```diff
@@ -114,18 +118,25 @@ async def test_charm_issues_metrics_for_abnormal_termination(
     assert workflow.create_dispatch(forked_github_branch, {"runner": app.name})
 
     await wait_for_workflow_to_start(
-        unit, workflow, branch=forked_github_branch, started_time=dispatch_time
+        unit,
+        workflow,
+        branch=forked_github_branch,
+        started_time=dispatch_time,
+        instance_helper=instance_helper,
     )
 
+    # Wait a bit to ensure pre-job script has been executed.
+    await sleep(5)
```

Comment:
What is needed for be waited for?
Would some form of readiness/health check be a more robust approach?

Reply:
I came across strange cases where the timestamp of RunnerStop was lower than that of RunnerStart. My suspicion is that the pre-job was executed after the post-job phase. I added a sleep to avoid this case.

Reply:
Have removed the sleep.